cpp/cmake/thirdparty/CUDF_FindBoost.cmake

```diff
@@ -0,0 +1,36 @@
+#=============================================================================
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+
+# Min version set to newest boost in Ubuntu bionic apt repositories
+set(CUDF_MIN_VERSION_Boost 1.65.0)
+
+# Don't look for a Boost CMake configuration file because it adds the
+# `-DBOOST_ALL_NO_LIB` and `-DBOOST_FILESYSTEM_DYN_LINK` compile defs
+set(Boost_NO_BOOST_CMAKE ON)
```

Comment:
I think we need options to control whether boost shared or static libraries are found / used as well?

Reply:
Did we have an option to statically link boost before? I don't recall seeing one.

Reply:
I believe it was reliant on `Boost_USE_STATIC_LIBS` from https://cmake.org/cmake/help/latest/module/FindBoost.html which should still work. Going to resolve this.